noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/ParameterInterpeter.java

```diff
@@ -0,0 +1,271 @@
+package edu.uci.ics.cloudberry.noah.crawler;
+
+import com.crawljax.browser.EmbeddedBrowser.BrowserType;
+import com.crawljax.core.CrawljaxException;
+import com.crawljax.core.configuration.CrawlRules;
+import com.google.common.base.Joiner;
+import org.apache.commons.cli.*;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.validator.routines.UrlValidator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+class ParameterInterpeter {
+
+	static final String HELP_MESSAGE =
+	        "java -jar crawljax-cli-version.jar theUrl theOutputDir";
+
+	private static final int SPACES_AFTER_OPTION = 3;
+	private static final int SPACES_BEFORE_OPTION = 5;
+	private static final int ROW_WIDTH = 80;
+
+	static final String VERSION = "version";
+	static final String VERBOSE = "verbose";
+	static final String HELP = "help";
+    static final String MAXSTATES = "maxstates";
+	static final String DEPTH = "depth";
+	static final String BROWSER = "browser";
+	static final String BROWSER_REMOTE_URL = "browserRemoteUrl";
+	static final String PARALLEL = "parallel";
+	static final String OVERRIDE = "override";
+	static final String CRAWL_HIDDEN_ANCHORS = "crawlHiddenAnchors";
+	static final String TIME_OUT = "timeout";
+	static final String WAIT_AFTER_RELOAD = "waitAfterReload";
+	static final String WAIT_AFTER_EVENT = "waitAfterEvent";
+	//static final String LOG_FILE = "log";
+	static final String CLICK = "click";
+	static final String DC = "dontclick";
+
+	private final Options options;
+	private final CommandLine parameters;
+
+	ParameterInterpeter(String args[]) throws ParseException {
+		this.options = getOptions();
+		this.parameters = new GnuParser().parse(options, args);
+	}
+
+	/**
+	 * Create the CML Options.
+	 * 
+	 * @return Options expected from command-line.
+	 */
+	private Options getOptions() {
```

Comment:
Yes you are right... We do not need the cmd options as we are making the
config file separate... I wanted confirmation from you regarding
"completely removing the cmd options".
On Jun 22, 2016 11:16 AM, "Jianfeng Jia" <notifications@github.com> wrote:

> In
> noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/ParameterInterpeter.java
> <https://github.com/ISG-ICS/cloudberry/pull/100#discussion_r68104741>:
>
> > +	static final String DC = "dontclick";
> > +
> > +	private final Options options;
> > +	private final CommandLine parameters;
> > +
> > +	ParameterInterpeter(String args[]) throws ParseException {
> > +		this.options = getOptions();
> > +		this.parameters = new GnuParser().parse(options, args);
> > +	}
> > +
> > +	/**
> > +	 * Create the CML Options.
> > +	 *
> > +	 * @return Options expected from command-line.
> > +	 */
> > +	private Options getOptions() {
>
> Do we still need those cmd options if using the config file?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ISG-ICS/cloudberry/pull/100/files/b541c7e50305cce36a85e40855e7ece1ef680670#r68104741>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARXIu_93qmpROZptvNQuM5JKPoslGRYCks5qOXo7gaJpZM4I8BJd>
> .
>